ethcore/src/snapshot/account.rs

```diff
@@ -62,53 +63,50 @@ impl CodeState {
 
 // walk the account's storage trie, returning an RLP item containing the
 // account properties and the storage.
-pub fn to_fat_rlp(acc: &BasicAccount, acct_db: &AccountDB, used_code: &mut HashSet<H256>) -> Result<Bytes, Error> {
+pub fn to_fat_rlps(acc: &BasicAccount, acct_db: &AccountDB, used_code: &mut HashSet<H256>, preferred_size: usize) -> Result<Vec<Bytes>, Error> {
+	const AVERAGE_BYTES_PER_STORAGE_ENTRY: usize = 47;
 	if acc == &ACC_EMPTY {
-		return Ok(::rlp::NULL_RLP.to_vec());
+		return Ok(vec![::rlp::NULL_RLP.to_vec()]);
 	}
 
 	let db = TrieDB::new(acct_db, &acc.storage_root)?;
 
-	let mut pairs = Vec::new();
+	let pair_chunks: Vec<Vec<_>> = db.iter()?.chunks(preferred_size / AVERAGE_BYTES_PER_STORAGE_ENTRY).into_iter().map(|chunk| chunk.collect()).collect();
```

Comment:
The problem is that RLP list size has to be specified in advance. You can't precisely estimate size after RLP serialization without doing the serialization.  The spec should not really put a restriction on chunks size, so I'll leave it it for future optimization (which can be done without breaking compatibility).